modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2109,20 +2196,26 @@ protected void addProperties(Map<String, Schema> properties, List<String> requir
         if (schema instanceof ComposedSchema) {
             ComposedSchema composedSchema = (ComposedSchema) schema;
 
-            for (Schema component : composedSchema.getAllOf()) {
-                addProperties(properties, required, component);
+            if (composedSchema.getAllOf() != null) {
+                for (Schema component : composedSchema.getAllOf()) {
+                    addProperties(properties, required, component);
+                }
             }
 
             if (schema.getRequired() != null) {
                 required.addAll(schema.getRequired());
             }
 
             if (composedSchema.getOneOf() != null) {
-                throw new RuntimeException("Please report the issue: Cannot process oneOf (Composed Scheme) in addProperties: " + schema);
+                for (Schema component : composedSchema.getOneOf()) {
+                    addProperties(properties, required, component);
+                }
```

Comment:
What should we do for these oneOf or anyOf cases?
Before we were throwing exceptions but that failed the below two test cases in testComposedSchemaOneOfDiscriminatorMap:
- FruitAnyOfDisc (a schema selected by oneOf contains only anyOf definition)
- FruitOneOfDisc (a schema selected by oneOf contains only oneOf definition)

Reply:
What we do here is important because we are loading the oneOf and anyOf properties into the composed schema required and optional vars signatures.
Should required properties in oneOf and anyOf schema properties be loaded in as required or optional vars in their parent composed schemas?
In my opinion they should be optional because any one oneOf or anyOf model could be omitted.

Reply:
@wing328 @jimschubert what do we want to do here?
- load all oneOf/anyOf required + optional properties into their composedSchema **required + optional** vars lists?
- load all oneOf/anyOf required + optional properties into their composedSchema **optional** vars list ONLY?